crates/uv/tests/python_pin.rs

```diff
@@ -229,6 +230,50 @@ fn python_pin_no_python() {
     "###);
 }
 
+#[test]
+fn python_pin_compatible_with_requires_python() -> anyhow::Result<()> {
+    let context: TestContext = TestContext::new_with_versions(&["3.11", "3.12"]);
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["iniconfig"]
+        "#,
+    )?;
+
+    uv_snapshot!(context.filters(), context.python_pin().arg("3.11"), @r###"
+        success: false
+        exit_code: 2
+        ----- stdout -----
+
+        ----- stderr -----
+        error: Pinned python version is incompatible with Requires-Python: >=3.12
```

Comment:
Maybe we should say "is incompatible with the workspace Python requirement"? I think there are other messages like this, we should be consistent but I'm not sure what they say.

Reply:
I didn't find a similar message elsewhere. But we could output the `pyproject.toml` path here?

"The pinned Python version is incompatible with the workspace's Python requirement ({requirement}) specified at {workspace_root}."

What do you think?

Reply:
I think I was thinking of this one

https://github.com/astral-sh/uv/blob/540ff2430267e504887cf34c2e30264f7836ef3f/crates/uv/tests/lock.rs#L1556

There's also this like... workspace / project display name difference at 

https://github.com/astral-sh/uv/blob/540ff2430267e504887cf34c2e30264f7836ef3f/crates/uv/src/commands/project/run.rs#L159-L169

we should probably match that terminology?

Displaying the path seems too verbose if it's present in `-v`, I think we could stick with something like 

> The requested Python version is incompatible with the [project/workspace]'s `Requires-Python` requirement of `...`

Wdyt?

Reply:
Makes sense, done.